Modules/Scripted/DICOMPlugins/DICOMScalarVolumePlugin.py

```diff
@@ -224,10 +230,40 @@ def examineFiles(self, files):
             for tag in subseriesTags:
                 value = slicer.dicomDatabase.fileValue(file, self.tags[tag])
                 value = value.replace(",", "_")  # remove commas so it can be used as an index
+
                 if tag not in subseriesValues:
                     subseriesValues[tag] = []
-                if not subseriesValues[tag].__contains__(value):
+
+                if tag == "imageOrientationPatient" or tag == "diffusionGradientOrientation":
+                    if value != "":
+                        directionCosine = [float(i) for i in value.split("\\")]
+                        a = math.sqrt(directionCosine[0]*directionCosine[0] + directionCosine[1]*directionCosine[1] + directionCosine[2]*directionCosine[2])
```

Comment:
Stylistically it's odd to mix a very descriptive variable name `directionCosine` and a meaningless single character variable name `a` in the same line, especially when the long one is repeated so much.

I think it would be cleaner to make a numpy array `directionCosines` and use `numpy.linalg.norm` and to give ` a name like `directionMagnitude`.

Reply:
I would use `np.linalg.norm(directionCosine)` instead, but I don't think we actually need normalization.

Reply:
you are right, comment applied

Reply:
> I would use `np.linalg.norm(directionCosine)` instead, but I don't think we actually need normalization.

it is true that imageOrientationPatient should be already normalized. However I have experienced it is not always the case. The load function in the scalar plugin normalize them as well.